src/uct/ib/dc/dc_mlx5.c

```diff
@@ -404,6 +404,12 @@ static int UCS_F_ALWAYS_INLINE
 uct_dc_mlx5_force_full_handshake(uct_dc_mlx5_iface_t *iface, int full_handshake_config)
 {
     if (full_handshake_config == UCS_CONFIG_AUTO) {
+        /* If RDMA_WRITE is not needed, can use half-handshake */
```

Comment:
in addition to this: if uct_ib_mlx5_iface_has_ar(), and full handshake is not enabled (for whatever reason), the iface should remove PUT capabilities from uct_iface_query()
otherwise seems nothing prevents UCP from doing PUT even if did not pass this as a needed feature